IdentityCore/src/cache/mac/MSIDMacACLKeychainAccessor.m

```diff
@@ -70,26 +70,43 @@ - (nullable instancetype)initWithTrustedApplications:(nullable NSArray *)trusted
         {
             return nil;
         }
-
-        // Note: Apple seems to recommend serializing keychain API calls on macOS in this document:
-        // https://developer.apple.com/documentation/security/certificate_key_and_trust_services/working_with_concurrency?language=objc
-        // However, it's not entirely clear if this applies to all keychain APIs.
-        // Since our applications often perform a large number of cache reads on mulitple threads, it would be preferable to
-        // allow concurrent readers, even if writes are serialized. For this reason this is a concurrent queue, and the
-        // dispatch queue calls are used. We intend to clarify this behavior with Apple.
-        //
-        // To protect the underlying keychain API, a single queue is used even if multiple instances of this class are allocated.
-        static dispatch_once_t s_once;
-        dispatch_once(&s_once, ^{
-            s_synchronizationQueue = dispatch_queue_create("com.microsoft.msidmackeychaintokencache", DISPATCH_QUEUE_CONCURRENT);
-        });
-
+        
         MSID_LOG_WITH_CTX(MSIDLogLevelInfo, nil, @"Init MSIDMacACLPersistentTokenCache");
     }
 
     return self;
 }
 
++ (void)setS_synchronizationQueue:(dispatch_queue_t)s_synchronizationQueue
+{
+    if (s_defaultSynchronizationQueue)
```

Comment:
How this prevents setting the queue multiple times?

Reply:
There is a ```return``` within if block

Reply:
Just updated